signature/signer.go

```diff
@@ -0,0 +1,45 @@
+package signature
+
+import (
+	"crypto"
+	"crypto/x509"
+	"fmt"
+)
+
+// Signer is used to sign bytes generated after creating signature envelope.
+type Signer interface {
+	Sign(digest []byte) ([]byte, error)
+	CertificateChain() ([]*x509.Certificate, error) // note: check signature first
+	KeySpec() (KeySpec, error)
+}
+
+type LocalSigner interface {
+	Signer
+	PrivateKey() crypto.PrivateKey
+}
+
+type signer struct {
+	keySpec KeySpec
+	key     crypto.PrivateKey
+	certs   []*x509.Certificate
+}
+
+func NewSigner(certs []*x509.Certificate, key crypto.PrivateKey) (Signer, error) {
```

Comment:
@binbin-li Should we make it
```suggestion
func NewLocalSigner(certs []*x509.Certificate, key crypto.PrivateKey) (LocalSigner, error) {
```

/ccc @chloeyin

Reply:
that does make more sense since we only need to create local signers.